### DISCUSSION TITLE
如何注入用户字段

### DISCUSSION THREAD

**Discussion by GSCSDSUN** [koishijs/koishi#257] [Q&A]
`User.extend(() => ({ coins: 114514 }))`
参数类型 function(): {coins: number} 不可分配给参数类型 Getter   
类型 function(): {coins: number} 不可分配给类型 <T extends Index>(type: T, id: string) => Partial<User>    
类型 {coins: number} 不可分配给类型 Partial<User> 
看wiki里就是这样写的
![@$DV4HFBR75ZL(`9`8)0NA4](https://user-images.githubusercontent.com/46085443/116834571-4f401580-abf1-11eb-9add-550a1d6bc028.png)
用userFields获取字段后 
[W] mysql SELECT `flag`,`authority`,`usage`,`id`,`name`,`onebot`,`coins` FROM `user` _user WHERE `onebot` = '112222' +5s
[W] session userinfo
    ER_BAD_FIELD_ERROR: Unknown column 'coins' in 'field list'   at MysqlDatabase.query (E:\桌面\杂物\ProjectE\koishi\node_modules\koishi-plugin-mysql\lib\index.js:15
9:19)

**Comment by NWYLZW** [comment]
如果可以你最好贴一下你的部分代码
最好还是把各个版本信息贴一下
不同的版本，可能有不同的坑。

**Comment by shigma** [comment]
mysql 的话要补一段：

```js
Database.extend('koishi-plugin-mysql', ({ tables }) => {
  tables.user.coins = 'int' // 如果是浮点型就是 float 或者 double 等等
})
```

**Reply by shigma** [reply]
这是因为 mysql 还需要制定字段的类型，不然没法建立数据模型

**Reply by shigma** [reply]
确实这是文档的欠缺
